crates/exex/exex/src/backfill.rs

```diff
@@ -185,41 +185,70 @@ where
         let chain = Chain::new(blocks, executor.finalize(), None);
         Ok(chain)
     }
+
+    /// Returns an iterator that executes each block in the range separately and yields the
+    /// resulting [`Block`] and [`ExecutionOutcome`].
+    pub fn into_single_blocks(
```

Comment:
I would prefer to have a separate `SingleBlockBackfillJob` struct that implements the `Iterator` trait, thoughts @mattsse ?

Reply:
same, I'd have dedicated iter types for this, impl trait will likely cause issues here in case the user wants `Send` for example

Reply:
got it, `SingleBlockBackfillJob` should be available through the same `BackfillJobFactory`?

Reply:
I see single block execution as a more niche use case, so let's only have an `into_` method on the `BackfillJob`

Reply:
can we still have an `into_single_blocks` method, and not just a `From` implementation?

Reply:
Sure, what's the reasoning behind not implementing the From? Isn't it more idiomatic?

Reply:
yep, let's leave it, but also have a separate method. The reason is that it's easier to write
```rust
let single_job = job.into_single_blocks();
```
instead of 
```rust
let single_job: SingleBlockBackfillJob<_, _, _> = job.into();
```